### PR TITLE
feat(suite-native): toggleable secute text passphrase input

### PR DESCRIPTION
**Pull request by PeKne**
## Description
- SecureTextInputField component created
  - this componet behaves as a TextIputField but allows to hide/show text content with a toggle
## Related Issue

Related to #12618

## Screenshots:

https://github.com/trezor/trezor-suite/assets/26143964/b87994db-db6c-47a7-b62b-7d4faaa5e585

### CODE REVIEW

**Comment by PeKne** [comment]
Commit [3661121](https://github.com/trezor/trezor-suite/pull/12727/commits/3661121c8010f79b34a6eda29aa6278b9a634db6) changes the `eyeSlash` from solid to outline:

https://github.com/trezor/trezor-suite/assets/26143964/4777e078-feb0-4329-92f1-efdf59bb4b4b

File: suite-native/forms/src/fields/SecureTextInputField.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+type ToggleSecureTextIconProps = {
+    onPress: () => void;
+    isTextHidden: boolean;
+};
+
+const ToggleSecureTextIcon = ({ onPress, isTextHidden }: ToggleSecureTextIconProps) => {
+    const iconName: IconName = isTextHidden ? 'eye' : 'eyeSlash';
+
+    return (
+        <Pressable onPress={onPress}>
```
Comment (juriczech) [inline suite-native/forms/src/fields/SecureTextInputField.tsx:19]: I would probably prefer using TouchableOpacity, but it's inside an input, so it's debatable. Feel free to leave it as it is, but some visual feedback onPress would be nicer, WDYT?

**Review by juriczech — APPROVED** [review]
LGTM, just wondering about the visual feedback on the icon, however that's not too important right now, I think.

Edit: I tested this on iOS (also tested max length), not android, but I see you tested Android so should be fine to merge.

File: suite-native/forms/src/fields/SecureTextInputField.tsx (hunk 1 above)
Comment (PeKne) [reply suite-native/forms/src/fields/SecureTextInputField.tsx:19]: I originally used `TouchableOpacity` but changed it to `Pressable` before pushing it. It felt weird that the eye button was animated, but the text changing to dots and back was not. So I decided to remove the animation to unify it. It feels somehow more natural, but it is a matter of taste for sure. Let's leave it as it is. I don't mind changing it later if Design or Product will complain.

**Pull request merged**